src/slam_toolbox_common.cpp

```diff
@@ -84,28 +182,44 @@ void SlamToolbox::configure()
 SlamToolbox::~SlamToolbox()
 /*****************************************************************************/
 {
+  scan_filter_.reset();
+
   for (int i = 0; i != threads_.size(); i++) {
     threads_[i]->join();
+    threads_[i].reset();
   }
-
-  smapper_.reset();
-  dataset_.reset();
+  threads_.clear();
+  // delete in reverse order of initialization
   closure_assistant_.reset();
   map_saver_.reset();
+  scan_holder_.reset();
   pose_helper_.reset();
   laser_assistant_.reset();
-  scan_holder_.reset();
   solver_.reset();
+
+  scan_filter_sub_.reset();
+  ssDesserialize_.reset();
+  ssSerialize_.reset();
+  ssPauseMeasurements_.reset();
+  ssMap_.reset();
+  sstm_.reset();
+  sst_.reset();
+  pose_pub_.reset();
+
+  tfB_.reset();
+  tfL_.reset();
+  tf_.reset();
+
+  dataset_.reset();
```

Comment:
Revert. THese must be reset before others -- in the order previously specified

Reply:
reverted :pray: [00a6215fd214c0403c2cf519f33be612a1ad0e08]